### PR TITLE
[pigeon] Add Flutter API integration tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
Adds `callFlutter*` wrappers for all of the recently-add Flutter API `echo*` methods in `core_tests.dart`, and adds integration tests driving them all.

This doesn't fix any of the exposed issues; they are all runtime rather than compile time, so I opted to file issues for each of them instead so they can be fixed on a per-generator basis (and so fixing them won't block getting the tests in place).

Part of https://github.com/flutter/flutter/issues/116374

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/CoreTests.gen.kt [hunk 1]
```diff
@@ -270,7 +270,22 @@ interface HostIntegrationCoreApi {
   /** Returns the passed string asynchronously. */
   fun echoAsyncString(aString: String, callback: (String) -> Unit)
   fun callFlutterNoop(callback: () -> Unit)
+  fun callFlutterEchoAllTypes(everything: AllTypes, callback: (AllTypes) -> Unit)
+  fun callFlutterSendMultipleNullableTypes(aNullableBool: Boolean?, aNullableInt: Long?, aNullableString: String?, callback: (AllNullableTypes) -> Unit)
```
Comment (tarrinneal) [inline packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/CoreTests.gen.kt:274]: Do we have a linter/formatter for kotlin? Is this something we can roll in to our tools?

File: packages/pigeon/platform_tests/test_plugin/ios/Classes/CoreTests.gen.swift [hunk 2]
```diff
@@ -281,7 +281,22 @@ protocol HostIntegrationCoreApi {
   /// Returns the passed string asynchronously.
   func echoAsyncString(aString: String, completion: @escaping (String) -> Void)
   func callFlutterNoop(completion: @escaping () -> Void)
+  func callFlutterEchoAllTypes(everything: AllTypes, completion: @escaping (AllTypes) -> Void)
+  func callFlutterSendMultipleNullableTypes(aNullableBool: Bool?, aNullableInt: Int32?, aNullableString: String?, completion: @escaping (AllNullableTypes) -> Void)
```
Comment (tarrinneal) [inline packages/pigeon/platform_tests/test_plugin/ios/Classes/CoreTests.gen.swift:285]: Same with swift

**Review by tarrinneal — APPROVED** [review]

File: packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/CoreTests.gen.kt (hunk 1 above)
Comment (stuartmorgan) [reply packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/CoreTests.gen.kt:274]: We don't have one; filed https://github.com/flutter/flutter/issues/118756

File: packages/pigeon/platform_tests/test_plugin/ios/Classes/CoreTests.gen.swift (hunk 2 above)
Comment (stuartmorgan) [reply packages/pigeon/platform_tests/test_plugin/ios/Classes/CoreTests.gen.swift:285]: Tracked by https://github.com/flutter/flutter/issues/41129; hopefully we'll have that fairly soon.

**Comment by stuartmorgan** [comment]
Per offline discussion the Java implementation could potentially be a little less verbose by using a shared generic result class. I'm going to go ahead and land it as-is since that's a minor detail in test-only code that shouldn't need to change, but I'll see if it's easy and if so do a follow-up later with it.

File: packages/pigeon/platform_tests/shared_test_plugin_code/lib/integration_tests.dart [hunk 3]
```diff
@@ -97,21 +97,11 @@ void runPigeonIntegrationTests(TargetGenerator targetGenerator) {
...
-      //         echoObject.mapWithAnnotations, genericAllTypes.mapWithAnnotations),
-      //     true);
-      // expect(
-      //     mapEquals(echoObject.mapWithObject, genericAllTypes.mapWithObject), true);
       expect(echoObject.anEnum, genericAllTypes.anEnum);
-    });
+    },
+        // TODO(stuartmorgan): Fix and re-enable.
+        // See https://github.com/flutter/flutter/issues/118726
+        skip: targetGenerator == TargetGenerator.kotlin);
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/pigeon/platform_tests/shared_test_plugin_code/lib/integration_tests.dart:104]: Oops, it looks like we're not running Kotlin tests in CI by mistake; the only `_integration_test` hit in the CI output is Swift. I'll look into that.

[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/pigeon/platform_tests/shared_test_plugin_code/lib/integration_tests.dart:104]: Oh right, that's https://github.com/flutter/flutter/issues/111505 🤦🏻

**Pull request merged**